L1Trigger/L1THGCal/plugins/ntuples/HGCalTriggerNtupleGenJet.cc

```diff
@@ -0,0 +1,95 @@
+#include "DataFormats/HepMCCandidate/interface/GenParticle.h"
+#include "DataFormats/JetReco/interface/GenJet.h"
+#include "DataFormats/JetReco/interface/GenJetCollection.h"
+
+#include "L1Trigger/L1THGCal/interface/HGCalTriggerNtupleBase.h"
+
+
+
+class HGCalTriggerNtupleGenJet : public HGCalTriggerNtupleBase
+{
+
+    public:
+        HGCalTriggerNtupleGenJet(const edm::ParameterSet& );
+
+        virtual void initialize(TTree&, const edm::ParameterSet&, edm::ConsumesCollector&&) override final;
+        virtual void fill(const edm::Event&, const edm::EventSetup& ) override final;
+
+    private:
+        virtual void clear() override final;
+
+        edm::EDGetToken genjet_token_;
+
+        int genjet_n_;
+        std::vector<float> genjet_energy_;
+        std::vector<float> genjet_pt_;
+        std::vector<float> genjet_eta_;
+        std::vector<float> genjet_phi_;
+
+        //std::vector<int>   gen_components_;
+
+};
+
+DEFINE_EDM_PLUGIN(HGCalTriggerNtupleFactory,
+        HGCalTriggerNtupleGenJet,
+        "HGCalTriggerNtupleGenJet" );
+
+
+HGCalTriggerNtupleGenJet::
+HGCalTriggerNtupleGenJet(const edm::ParameterSet& conf):HGCalTriggerNtupleBase(conf)
+{
+}
+
+void
+HGCalTriggerNtupleGenJet::
+initialize(TTree& tree, const edm::ParameterSet& conf, edm::ConsumesCollector&& collector)
+{
+
+    genjet_token_ = collector.consumes<reco::GenJetCollection>(conf.getParameter<edm::InputTag>("GenJets"));
+    tree.Branch("genjet_n", &genjet_n_, "gen_n/I");
+    tree.Branch("genjet_energy", &genjet_energy_);
+    tree.Branch("genjet_pt", &genjet_pt_);
+    tree.Branch("genjet_eta", &genjet_eta_);
+    tree.Branch("genjet_phi", &genjet_phi_);
+
+}
+
+void
+HGCalTriggerNtupleGenJet::
+fill(const edm::Event& e, const edm::EventSetup& es)
+{
+    edm::Handle<std::vector<reco::GenJet> > genjets_h;
```

Comment:
For consistency with the above line 48, better to use `reco::GenJetCollection` typedef